Add DJM callout to Spark documentation (#18157)
DataDog/integrations-core: spark/README.md

# Spark Check

<div class="alert alert-warning">
<a href="https://docs.datadoghq.com/data_jobs/">Data Jobs Monitoring</a> helps you observe, troubleshoot, and cost-optimize your Spark and Databricks jobs and clusters.<br/><br/>

> This page is limited to documentation for ingesting Spark metrics and logs. 

Reviewer: ```suggestion
This page only documents how to ingest Spark metrics and logs. 
```
